jaxrs/resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/proxy/ClientProxy.java

```diff
@@ -43,7 +52,21 @@ public Object invoke(Object o, Method method, Object[] args)
       MethodInvoker clientInvoker = methodMap.get(method);
       if (clientInvoker == null)
       {
-         if (method.getName().equals("equals"))
+	      if (isDefault(method)) {
+		      // Call default methods as-is.
+		      // Approach from https://rmannibucau.wordpress.com/2014/03/27/java-8-default-interface-methods-and-jdk-dynamic-proxies/
```

Comment:
Note that this is likely to break in Java 9 (or so we've been told)

Reply:
The new `privateLookupIn` method fixes this, although it is not available with 8 so you have to do it reflectively: https://github.com/jdbi/jdbi/pull/1066